biosdk-services/src/main/java/io/mosip/biosdk/services/impl/spec_1_0/BioSdkServiceProviderImpl_V_1_0.java

```diff
@@ -157,6 +187,30 @@ public Object segment(RequestDto request) {
                     segmentRequestDto.getModalitiesToSegment(),
                     segmentRequestDto.getFlags()
             );
+            boolean isValid = true;
+            //Validation
+            if (response != null && response.getResponse() != null)
+            {
+        		BiometricRecord biometricRecord = (BiometricRecord)response.getResponse();
+                //Should have Valid BIR Segments
+        		List<BIR> segments = biometricRecord.getSegments();
+        		if (segments == null || segments.size() == 0)
+        		{
+        			isValid = false;
+        		}
+            }
+            else
+            {
+    			isValid = false;
+            }
+            if (!isValid)
+            {
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response Code  :: ", response.getStatusCode());        			
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response Status :: ", response.getStatusMessage());        			
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Request :: ", request.getRequest());
+    	        logger.debug(LOGGER_SESSIONID, LOGGER_IDTYPE, "extractTemplate: Response :: ", response.getResponse());
+				response.setStatusCode(ResponseStatus.UNKNOWN_ERROR.getStatusCode());
+            } 
```

Comment:
do we need this validation and debug for segement also ?

Reply:
for segments not required.. we do not use it